pkg/syncer/admissionhandler/admissionhandler.go

```diff
@@ -0,0 +1,257 @@
+	}
+	if k8s == nil {
+		k8s, err = commonco.GetContainerOrchestratorInterface(ctx, common.Kubernetes, cfg.FeatureStatesConfig)
+		if err != nil {
+			log.Errorf("failed to get k8s interface. err: %v", err)
+			return err
+		}
+	}
+	if k8s.IsFSSEnabled(ctx, common.CSIMigration) {
+		certs, err := tls.LoadX509KeyPair(cfg.WebHookConfig.CertFile, cfg.WebHookConfig.KeyFile)
+		if err != nil {
+			log.Errorf("failed to load key pair. certFile: %q, keyFile: %q err: %v", cfg.WebHookConfig.CertFile, cfg.WebHookConfig.KeyFile, err)
+			return err
+		}
+		if cfg.WebHookConfig.Port == "" {
+			cfg.WebHookConfig.Port = defaultWebhookServerPort
+		}
+		server = &http.Server{
+			Addr:      fmt.Sprintf(":%v", cfg.WebHookConfig.Port),
+			TLSConfig: &tls.Config{Certificates: []tls.Certificate{certs}},
+		}
+		// define http server and server handler
+		mux := http.NewServeMux()
+		mux.HandleFunc("/validate", validationHandler)
+		server.Handler = mux
+
+		// start webhook server
+		log.Debugf("starting webhook server on port: %v", cfg.WebHookConfig.Port)
+		go func() {
+			if err = server.ListenAndServeTLS(cfg.WebHookConfig.CertFile, cfg.WebHookConfig.KeyFile); err != nil {
+				if err == http.ErrServerClosed {
+					log.Info("webhook server stopped")
+				} else {
+					log.Fatalf("failed to listen and serve webhook server. err: %v", err)
+				}
+			}
+		}()
+		log.Info("webhook server started")
+		watchConfigChange()
+		<-stopCh
+		return nil
+	}
+	return errors.New("can't start webhook. no features are enabled which requires webhook")
+}
+
+// restartWebhookServer stops the webhook server and start webhook using updated config
+func restartWebhookServer(ctx context.Context) error {
+	log := logger.GetLogger(ctx)
+	cfg, err := getWebHookConfig(ctx)
+	if err != nil {
+		log.Errorf("failed to get webhook config. err: %v", err)
+	}
+	log.Debugf("webhook config: %v", cfg)
+	log.Info("shutting down webhook server gracefully...")
+	err = server.Shutdown(context.Background())
+	if err != nil {
+		log.Errorf("failed to get shutdown webhook server. err: %v", err)
+		return err
+	}
+	log.Info("restarting webhook server...")
+	return StartWebhookServer(ctx)
+}
+
+// validationHandler is the handler for webhook http multiplexer to help validate resources
+// depending on the URL validation of AdmissionReview will be redirected to appropriate validation function
+func validationHandler(w http.ResponseWriter, r *http.Request) {
+	var body []byte
+	ctx, log := logger.GetNewContextWithLogger()
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+	if len(body) == 0 {
+		log.Error("received empty request body")
+		http.Error(w, "received empty request body", http.StatusBadRequest)
+		return
+	}
+	log.Debugf("Received request")
+	// verify the content type is accurate
+	contentType := r.Header.Get("Content-Type")
+	if contentType != "application/json" {
+		log.Errorf("Content-Type=%s, expect application/json", contentType)
```

Comment:
Nit: Fix error message to not start with init cap.

Reply:
done.